### ISSUE TITLE
Feature Request: Play playists in reverse

### ISSUE THREAD

**Issue by SexyNick** [yuliskov/SmartTubeNext#2648]
A lot of YouTubers will arrange their content in playlists going from newest > oldest. Being able to autoplay them in reverse (and thus chronological) order would be pretty useful and simple to implement.

**Comment by Overwatching** [comment]
It would be nice to have a toggle setting for each playlist that controls the order it get's played in.

There is a workaround for now though.

1. Go to the playlist section.
2. Highlight the playlist you want to change.
3. Menu button.
4. Sort playlist.
5. Pick the order you want it to be played.

That will do a one-time(?) sort of that playlist to the order you selected.

This works well for playlists that are already "done", but for actively updated lists new titles to [order creator uses].
So, if the creator puts new videos on-top, then new entries will still come first, and you will need to re-sort.
Example of a sorted list, where creator adds a new video to the top...

- Ep. 20
- Ep. 1
- Ep. 2
- Ep. 3
- ...

The opposite will be true for a list like "Watch Later", where new entries are added to the end by YouTube.
If you have your list sorted [Latest Added First] it will still add new entries to the end. Ex...

- Ep. 6
- Ep. 5
- Ep. 4
- ...
- Ep. 7

Resorting will go through everything and set it to the correct order.
Not ideal, but still usable. For me at least.

**Comment by SexyNick** [comment]
@Overwatching Unfortunately, you can only sort playlists you own. If you add someone else's playlist to your playlist section, it will throw an error if you try to sort it.

**Comment by Overwatching** [comment]
@SexyNick That doesn't actually make sense...
The sorting is being done on the client device.

I have this on 3 TVs and doing a sort on TV1 does nothing to the order of TV2 or TV3. Nor does it change the order on any of my computers.

I could have sworn it worked on all playlists...

**Comment by Abysselene** [comment]
Indeed it doesn't work on playlist referenced from public one. I agree it should, it's just local sort.
A temporary workaround to duplicate a public playlist into your private one where you'll be able to sort it : https://www.reddit.com/r/youtube/s/M8KYGfYIPz
With this tip where you change the www from your playlist into music. You can add to a new or existing playlist from a playlist without having to pick the videos one by one.

**Comment by SexyNick** [comment]
Why do people keep saying playlists are saved locally? They're 100% server-side.

**Comment by Overwatching** [comment]
> Why do people keep saying playlists are saved locally? They're 100% server-side.

Because at some level IT IS stored locally?

Some information about that list, or a copy of it, or SOMETHING is local.

If I set a different watch order on the TV, which I can only do for a playlist I "own", that list does not change on my desktop.
That tells me there MUST be some value stored somewhere locally, or those changes could not persist, which they do, unless the base playlist was changed, which it is not.

**Comment by SexyNick** [comment]
> If I set a different watch order on the TV, which I can only do for a playlist I "own", that list does not change on my desktop. 

That's not true. The playlist's order will be changed everywhere if sorted via SmartTube's interface or through any other method/client.

**Comment by Overwatching** [comment]
> That's not true....

Dunno what to tell you.
I just did it.

- Set a playlist to play in reverse order on TV.
- Shows as original order on desktop.
- Add a video to the end of playlist on computer.
- Shows as first item on tv.

**Comment by SexyNick** [comment]
I did it before writing the comment and got the opposite result. Did you refresh the playlist page after sorting it on SmartTube? It's not going use a websocket or anything to update it on all devices if that's what you're expecting.
